test/e2e/traces_noisy_span_filter_test.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/kyma-project/telemetry-manager/test/testkit/verifiers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kyma-project/telemetry-manager/test/testkit/periodic"
+)
+
+var _ = Describe("Traces Noisy Span Filter", Label("tracing"), func() {
+
+	const (
+		mockBackendName = "traces-filter-receiver"
+		mockNs          = "traces-filter-test"
+	)
+
+	var (
+		pipelineName string
+		urls         = urlprovider.New()
+	)
+
+	makeResources := func() []client.Object {
+		var objs []client.Object
+
+		objs = append(objs, kitk8s.NewNamespace(mockNs).K8sObject())
+
+		mockBackend := backend.New(mockBackendName, mockNs, backend.SignalTypeTraces, backend.WithPersistentHostSecret(true))
+		objs = append(objs, mockBackend.K8sObjects()...)
+		urls.SetMockBackendExport(mockBackend.Name(), mockBackend.TelemetryExportURL(proxyClient))
+
+		pipeline := kittrace.NewPipeline(fmt.Sprintf("%s-pipeline", mockBackend.Name())).
+			WithOutputEndpointFromSecret(mockBackend.HostSecretRef()).
+			Persistent(isOperational())
+		pipelineName = pipeline.Name()
+		objs = append(objs, pipeline.K8sObject())
+
+		urls.SetOTLPPush(proxyClient.ProxyURLForService(
+			kitkyma.SystemNamespaceName, "telemetry-otlp-traces", "v1/traces/", ports.OTLPHTTP),
+		)
+
+		traceGatewayExternalService := kitk8s.NewService("telemetry-otlp-traces-external", kitkyma.SystemNamespaceName).
+			WithPort("grpc-otlp", ports.OTLPGRPC).
+			WithPort("http-metrics", ports.Metrics)
+		urls.SetMetrics(proxyClient.ProxyURLForService(
+			kitkyma.SystemNamespaceName, "telemetry-otlp-traces-external", "metrics", ports.Metrics))
+
+		objs = append(objs, traceGatewayExternalService.K8sObject(kitk8s.WithLabel("app.kubernetes.io/name", kitkyma.TraceGatewayBaseName)))
+		return objs
+	}
+
+	Context("When noisy span present", Ordered, func() {
+
+		BeforeAll(func() {
+			k8sObjects := makeResources()
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should be able to get trace gateway metrics endpoint", Label(operationalTest), func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(urls.Metrics())
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+			}, periodic.EventuallyTimeout, periodic.DefaultInterval).Should(Succeed())
+		})
+
+		It("Should have a running pipeline", Label(operationalTest), func() {
+			verifiers.TracePipelineShouldBeRunning(ctx, k8sClient, pipelineName)
+		})
+
+		It("Should verify end-to-end trace delivery", Label(operationalTest), func() {
+			traceID, spanIDs, attrs := kittraces.MakeAndSendTraces(proxyClient, urls.OTLPPush())
+			verifiers.TracesShouldBeDelivered(proxyClient, urls.MockBackendExport(mockBackendName), traceID, spanIDs, attrs)
+		})
+
+		It("Should filter noisy traces and spans", Label(operationalTest), func() {
```

Comment:
Maybe let's split this to one `It` per test case to make it better readable. Maybe use the `By` for the step of sending traces.

Reply:
Discussion with Stas abut this point, we come to the conclusion, doing dedicated `It` per test case will put more pressure on the API Server

Reply:
I don't understand that point. It is only about the organization of the test cases. The API server calls should not change.

Reply:
each test case moved in own execution block